hooks/node_handlers/base_render_handler.py

```diff
@@ -0,0 +1,372 @@
+import sgtk
+
+import hou
+
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class BaseRenderNodeHandler(HookBaseClass):
+    """
+    Base class for all render output nodes.
+    """
+
+    SGTK_PASS_NAME = "sgtk_pass_name"
+
+    # aovs
+    AOV_COUNT = ""
+    AOV_NAME_TMPL = "{}"
+    AOV_FILE_TMPL = "{}"
+    AOV_USE_FILE_TMPL = "{}"
+
+    # archives
+    ARCHIVE_ENABLED = ""
+    ARCHIVE_OUTPUT = ""
+
+    # templates
+    AOV_WORK_TEMPLATE = "aov_work_template"
+    AOV_PUBLISH_TEMPLATE = "aov_publish_template"
+    ARCHIVE_WORK_TEMPLATE = ""
+    ARCHIVE_PUBLISH_TEMPLATE = ""
+
+    # strings
+    AOV_ERROR = ""
+
+    def _get_template_fields_from(self, node, additional_fields={}):
+        """
+        Get shotgun template fields from the given node and update with any additional
+        fields.
+
+        :param node: A :class:`hou.Node` instance.
+        :param dict additional_fields: Any fields to override with.
+
+        :returns: The template fields updated with the additional_fields
+        """
+        output_parm = node.parm(self.OUTPUT_PARM)
+        file_path = output_parm.unexpandedString()
+        fields = self.get_work_template(node).validate_and_get_fields(file_path)
+        if not fields:
+            mesage = 'Can not extract Shotgun fields from "{}": "{}"'
+            raise sgtk.TankError(mesage.format(output_parm.path(), file_path))
+        if "SEQ" in fields:
+            fields["SEQ"] = "FORMAT: $F"
+        fields.update(additional_fields)
```

Comment:
Part of above parameter comment:

```suggestion
        fields.update(additional_fields or {})
```